snippets/nix.json

```
@@ -82,6 +82,46 @@
         ],
         "description": "Nixpkgs' stdenv.mkDerivation template"
     },
+    "pkgs.mkShell": {
+        "prefix": "mks",
+        "body": [
+            "{pkgs ? import <nixpkgs> {}, ...}:",
+            "pkgs.mkShell {",
+            "  shellHook = ''",
+            "    $1",
+            "  '';",
+            "",
+            "  env = {",
+            "    $2",
+            "  };",
+            "",
+            "  packages = [",
+            "    $3",
+            "  ];",
+            "}"
```

```suggestion
            "pkgs.mkShell {",
            "  shellHook = ''",
            "    $1",
            "  '';",
            "",
            "  env = {",
            "    $2",
            "  };",
            "",
            "  packages = [",
            "    $3",
            "  ];",
            "};"
```

for this part I would make it both nix and flake compatible since it would be better for both compatibility and a user is expected to write `pkgs.mkShell` in a `shell.nix` file or in `devShells.default`

---

Yeah, makes sense, but then I'd need to write the preamble manually every single time I write a `shell.nix`...

---

Probably best to split it into two templates in that case.